syntax="proto2";
package InGamePacket;

message C_Move{
	required uint32 uid=1;
	required float x=2;
	required float y=3;
	required float direction=4;
	required float velocity=5;
}

message S_Move{
	required uint32 uid=1;
	required float x=2;
	required float y=3;
	required float direction=4;
	required float velocity=5;
}

message C_Stop{
	required uint32 uid=1;
	required float x=2;
	required float y=3;
}

message S_Stop{
	required uint32 uid=1;
	required float x=2;
	required float y=3;
}

message C_Shoot{
	required uint32 uid=1;
	required float x=2;
	required float y=3;
	required float damage=4;
	required float velocity=5;
	required float direction=6;
}

message S_Shoot{
	required uint32 uid=1;
	required float x=2;
	required float y=3;
	required float damage=4;
	required float velocity=5;
	required float direction=6;
}